PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/springcloud/SpringCloudAppPropertyView.java

```diff
@@ -0,0 +1,681 @@
+                this.model.setJavaVersion(settings.runtimeVersion().toString());
+                this.model.setJvmOptions(settings.jvmOptions());
+                this.model.setCpu(settings.cpu());
+                this.model.setMemoryInGB(settings.memoryInGB());
+                if (deploy.properties().instances() != null) {
+                    this.model.setDownInstanceCount((int) deploy.properties().instances().stream().filter(
+                        t -> StringUtils.equalsIgnoreCase(t.discoveryStatus(), "DOWN")).count());
+                    this.model.setUpInstanceCount(deploy.properties().instances().size() - this.model.getDownInstanceCount());
+                    this.model.setInstance(deploymentResourceInner.properties().instances().stream().map(t -> {
+                        SpringAppInstanceViewModel instanceViewModel = new SpringAppInstanceViewModel();
+                        instanceViewModel.setName(t.name());
+                        instanceViewModel.setStatus(t.status());
+                        instanceViewModel.setDiscoveryStatus(t.discoveryStatus());
+                        return instanceViewModel;
+                    }).collect(Collectors.toList()));
+                } else {
+                    this.model.setUpInstanceCount(0);
+                    this.model.setDownInstanceCount(0);
+                }
+                // env variable
+                this.model.setEnvironment(settings.environmentVariables());
+            } else {
+                this.model.setUpInstanceCount(0);
+                this.model.setDownInstanceCount(0);
+            }
+            // public url
+            this.model.setEnablePublicUrl(app.properties().publicProperty());
+            if (this.model.isEnablePublicUrl()) {
+                this.model.setPublicUrl(app.properties().url());
+            }
+
+            this.model.setClusterName(SpringCloudIdHelper.getClusterName(this.appId));
+            this.model.setAppName(app.name());
+
+            // button enable
+            DeploymentResourceStatus status = deploy == null ? DeploymentResourceStatus.UNKNOWN : deploy.properties().status();
+            boolean stopped = DeploymentResourceStatus.STOPPED.equals(status);
+            boolean unknown = DeploymentResourceStatus.UNKNOWN.equals(status);
+            this.model.setCanStop(!stopped && !unknown);
+            this.model.setCanStart(stopped);
+            this.model.setCanReStart(!stopped && !unknown);
+            // status
+            this.model.setStatus(status.toString());
+            this.setData(model);
+        } catch (AzureExecutionException e) {
+            ApplicationManager.getApplication().invokeLater(() -> {
+                PluginUtil.showErrorNotificationProject(project, "Cannot binding data to Spring Cloud property view.", e.getMessage());
+            });
+        }
+    }
+
+    private void setData(SpringAppViewModel model) throws AzureExecutionException {
```

Comment:
setData -> refreshUI? As setData seems more likely an Model operation

Reply:
setData is changed